### PR TITLE
zh-translation: content/zh/docs/concepts/architecture/controller.md

### PR DESCRIPTION
**Pull request by helight**
ref:https://github.com/k8smeetup/website-tasks/issues/3654

### CODE REVIEW

File: content/zh/docs/concepts/architecture/controller.md [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+
+Kubernetes 自带有一组内置的控制器，运行在 {{< glossary_tooltip term_id="kube-controller-manager" >}} 内。这些内置的控制器提供了重要的核心功能。
+
+Deployment 控制器和 Job 控制器是 Kubernetes 内置控制器的典型例子。Kubernetes 运行一个弹性的控制平面，所以如果任意内置控制器失败了，控制平面的另外一部分会接替它的工作。
+
+You can find controllers that run outside the control plane, to extend Kubernetes.
+Or, if you want, you can write a new controller yourself.
+You can run your own controller as a set of Pods,
+or externally to Kubernetes. What fits best will depend on what that particular
+controller does.
```
Comment (sftim) [inline content/zh/docs/concepts/architecture/controller.md]: Should this be commented out?

Comment (tanjunchen) [reply content/zh/docs/concepts/architecture/controller.md]: it should be commented.

File: content/zh/docs/concepts/architecture/controller.md [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+
+一个控制器至少追踪一种类型的 Kubernetes 资源。这些[对象](/docs/concepts/overview/working-with-objects/kubernetes-objects/)有一个代表期望状态的指定字段。正对这种资源的控制器就是要使他的当前状态接近与期望状态。
+
+控制器可能会自行执行操作；在 Kubernetes 中更常见的是一个控制器会发送信息给 {{< glossary_tooltip text="API 服务" term_id="kube-apiserver" >}}，这会有副作用。看下面这个例子。
+
+{{< comment >}}
+一些内置的控制器，比如命名空间控制器，针对没有指定命名空间的对象。简单期间，这篇文章没有详细介绍这些细节。
+{{< /comment >}}
+
+<!--
```
Comment (tanjunchen) [inline content/zh/docs/concepts/architecture/controller.md]: the comment is too long , be shorter for review.

File: content/zh/docs/concepts/architecture/controller.md (hunk 1 above)
Comment (helight) [reply content/zh/docs/concepts/architecture/controller.md]: thank you, those line should delete.

File: content/zh/docs/concepts/architecture/controller.md (hunk 2 above)
Comment (helight) [reply content/zh/docs/concepts/architecture/controller.md]: ok, got.

Comment (tengqm) [reply content/zh/docs/concepts/architecture/controller.md]: 简单期间？

File: content/zh/docs/concepts/architecture/controller.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+When you set the temperature, that's telling the thermostat
+about your *desired state*. The actual room temperature is the
+*current state*. The thermostat acts to bring the current state
+closer to the desired state, by turning equipment on or off.
+-->
+
+在机器人技术和自动化中，控制循环是一个控制系统状态的不终止的循环。
+
+这是一个控制循环的例子：房间里的温度自动调节器。
```
Comment (tengqm) [inline content/zh/docs/concepts/architecture/controller.md]: control loop 控制环

File: content/zh/docs/concepts/architecture/controller.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+当你设置了温度，告诉了温度自动调节器你*期望状态*。房间的实际温度是*当前状态*。通过对设备的开关控制，温度自动调节器让其当前状态接近期望状态。
```
Comment (tengqm) [inline content/zh/docs/concepts/architecture/controller.md]:
```suggestion
当你设置了温度，告诉了温度自动调节器你的*期望状态*。房间的实际温度是*当前状态*。通过对设备的开关控制，温度自动调节器让其当前状态接近期望状态。
```

File: content/zh/docs/concepts/architecture/controller.md [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+Some built-in controllers, such as the namespace controller, act on objects
+that do not have a spec. For simplicity, this page omits explaining that
+detail.
+{{< /comment >}}
+-->
+## 控制器模式 {#controller-pattern}
+
+一个控制器至少追踪一种类型的 Kubernetes 资源。这些[对象](/docs/concepts/overview/working-with-objects/kubernetes-objects/)有一个代表期望状态的指定字段。正对这种资源的控制器就是要使他的当前状态接近与期望状态。
+
+控制器可能会自行执行操作；在 Kubernetes 中更常见的是一个控制器会发送信息给 {{< glossary_tooltip text="API 服务" term_id="kube-apiserver" >}}，这会有副作用。看下面这个例子。
```
Comment (tengqm) [inline content/zh/docs/concepts/architecture/controller.md]: API server -> API 服务器

File: content/zh/docs/concepts/architecture/controller.md [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+Pods.
+Other components in the
+{{< glossary_tooltip text="control plane" term_id="control-plane" >}}
+act on the new information (there are new Pods to schedule and run),
+and eventually the work is done.
+-->
+
+### 通过 API 服务来控制 {#control-via-API-server}
+
+{{< glossary_tooltip term_id="job" >}} 控制器是一个 Kubernetes 内置控制器的例子。内置控制器管理通过和集群 API 服务交互来管理状态。
```
Comment (tengqm) [inline content/zh/docs/concepts/architecture/controller.md]: 第二句不通

File: content/zh/docs/concepts/architecture/controller.md [hunk 7]
```diff
@@ -0,0 +1,252 @@
...
+
+As long as the controllers for your cluster are running and able to make
+useful changes, it doesn't matter if the overall state is or is not stable.
+-->
+
+## 期望状态与当前状态 {#desired-vs-current}
+
+Kubernetes 采用了系统的云原生视图，并且可以处理持续的变化。
+
+在任务执行时，集群随时都可能被修改，并且控制循环会自动的修复故障。这意味着很可能集群永远不会达到稳定状态。
```
Comment (tengqm) [inline content/zh/docs/concepts/architecture/controller.md]: 控制环

File: content/zh/docs/concepts/architecture/controller.md (hunk 3 above)
Comment (helight) [reply content/zh/docs/concepts/architecture/controller.md]: accept

File: content/zh/docs/concepts/architecture/controller.md (hunk 4 above)
Comment (helight) [reply content/zh/docs/concepts/architecture/controller.md]: accept

File: content/zh/docs/concepts/architecture/controller.md (hunk 5 above)
Comment (helight) [reply content/zh/docs/concepts/architecture/controller.md]: accept

File: content/zh/docs/concepts/architecture/controller.md (hunk 6 above)
Comment (helight) [reply content/zh/docs/concepts/architecture/controller.md]: 应该是：内置控制器通过和集群 API 服务器交互来管理状态。

File: content/zh/docs/concepts/architecture/controller.md (hunk 7 above)
Comment (helight) [reply content/zh/docs/concepts/architecture/controller.md]: accept

**Pull request merged**